ssh_tunnel.go

```diff
@@ -42,36 +43,71 @@ func (tunnel *SSHTunnel) Start() error {
 }
 
 func (tunnel *SSHTunnel) forward(localConn net.Conn) {
-	serverConn, err := ssh.Dial("tcp", tunnel.Server.String(), tunnel.Config)
-	if err != nil {
-		tunnel.logf("server dial error: %s", err)
-		return
+	var serverConn *ssh.Client
+	var serverErr interface{}
+
+	var retry int
+	for {
+
+		retry ++
+		serverConn, serverErr = ssh.Dial("tcp", tunnel.Server.String(), tunnel.Config)
+		if serverErr != nil {
+			tunnel.logf("server dial error: %s", serverErr)
+		} else {
+			break
+		}
+	}
+
+	if retry > 1 {
+		tunnel.logf("Retry server: %s", strconv.Itoa(retry))
```

Comment:
There are a few things I don't like about this:

1. This is unlimited retry. There must be a way to configure it or give it a save default (such as 3 times).
2. It's repeated code, the same thing happens below. The retry mechanism can be moved to be a method.
3. There's no need to print the number of failures if you have already logged each failure.